src/MongoDB.Driver.Core/Core/Operations/ChangeStreamCursor.cs

```diff
@@ -181,5 +268,29 @@ private void ProcessBatch(bool hasMore)
                 _current = null;
             }
         }
+
+        private IAsyncCursor<RawBsonDocument> ResumeCursor(IAsyncCursor<RawBsonDocument> cursor, CancellationToken cancellationToken)
+        {
+            _changeStreamOperation = _changeStreamOperation.WithResumeOptions(_documentResumeToken, _postBatchResumeToken, _initialOperationTime);
+            var newCursor = _changeStreamOperation.Resume(_binding, cancellationToken);
+            cursor.Dispose();
+            return newCursor;
```

Comment:
I don't think we need to create a new operation for Resume. It takes a lot of code to clone the operation and we might make mistakes now or in the future when cloning. We can just reconfigure the resume values of the existing operation.

I would rename and simplify this method to:

    private IAsyncCursor<RawBsonDocument> Resume(CancellationToken cancellationToken)
    {
        ReconfigureOperationResumeValues();
        return _changeStreamOperation.Resume(_binding, cancellationToken);
    }

Where ReconfigureOperationResumeValues is:

    private void ReconfigureOperationResumeValues()
    {
        var resumeValues = GetResumeValues();
        _changeStreamOperation.ResumeAfter = resumeValues.ResumeAfter;
        _changeStreamOperation.StartAfter = resumeValues.StartAfter;
        _changeStreamOperation.StartAtOperationTime = resumeValues.StartAtOperationTime;
    }

Reply:
Done